src/DependencyInjection/services.xml

```diff
@@ -129,5 +129,11 @@
             <argument key="$languageRepository" type="service" id="language.repository" />
         </service>
 
+        <service id="PayonePayment\Components\CustomerDataPersistor\CustomerDataPersistor" autowire="true">
+            <argument key="$orderAddressRepository" type="service" id="order_address.repository" />
+            <argument key="$customerRepository" type="service" id="customer.repository" />
+            <argument key="$customerAddressRepository" type="service" id="customer_address.repository" />
```

Comment:
If you are using “autowire” anyway, you can actually omit the repository declarations, as autowiring should also work by default for these.

Reply:
i know. But in the past Shopware does have a few problems with injecting the repositories, so i never used this in production. 
my suggestion is to keep it in the backlog and refactoring this on the next SW 6.7 release.